src/Event/DefaultRoleEventInterface.php

```diff
@@ -95,11 +104,18 @@ public function deleteRole($name);
    * Returns whether or not the given role exists.
    *
    * @param string $name
-   *   The name of the role for which to verify the existance.
+   *   The name of the role for which to verify the existence.
    *
    * @return bool
    *   TRUE if the role exists, FALSE otherwise.
    */
   public function hasRole($name);
 
+  /**
+   * Resets the internal static cache.
+   *
+   * Call this before dispatching the event.
+   */
+  public function reset();
```

Comment:
huh? This seems like a weird thing to have to do if you're dispatching an event?